templates/conf.d/map.epp

```diff
@@ -0,0 +1,31 @@
+<%- |
+  Optional[String] $default,
```

Comment:
I think optional doesn't make sense here unless you give the variable a default value on `undef`?

Reply:
The default `undef` comes from the defined resource parameters.. So it is truly optional which is tested later in EPP (See `<% if $default { -%>` below)

Reply:
I can add `undef` here as default value too if it'll be less confusing